### PR TITLE
fix: Display Avatar Preference not hiding avatars on messages

### LINKED ISSUE #26741: Display Avatars switch:  unable to disable showing avatars in messages
User Preference:
Display Avatars: Hides avatars on the channels, showing only the users' usernames.

This setting since version 5 does not disable the viewing of user's avatars in messages.

### PR DESCRIPTION
**Pull request by nishant23122000**
## Issue(s)
Closes #26741

## Steps to test or reproduce

- Go to the user account
- Go to preferences
- In Message Section use the toggle button to disable or enable the **Display avatar**

You will able to toggle it, but if it's enabled, messages are rendering along with the user avatar.

### CODE REVIEW

**Comment by nishant23122000** [comment]
This same issue happens with a The hide username also. before that, I'm Waiting for the reviewer to check whether it's the right approach to do or not.

**Comment by dudanogueira** [comment]
Hi! There is at least 3 issues reagarding the real name issue areound:
https://github.com/RocketChat/Rocket.Chat/issues/23051
https://github.com/RocketChat/Rocket.Chat/issues/21480
https://github.com/RocketChat/Rocket.Chat/issues/21129

Indeed it needs more discussion. 
Thanks for this PR!! I'll put it with our team.

File: apps/meteor/client/views/room/MessageList/components/Message.tsx [hunk 1]
```diff
@@ -29,6 +30,7 @@ const Message: FC<{
 		actions: { openUserCard },
 	} = useMessageActions();
 
+	const showUserAvatar = !!useUserPreference<boolean>('displayAvatars');
```
Comment (debdutdeb) [inline apps/meteor/client/views/room/MessageList/components/Message.tsx:33]:
```suggestion
	const showUserAvatar = useUserPreference<boolean>('displayAvatars');
```
No need to convert.

**Review by debdutdeb — CHANGES_REQUESTED** [review]

File: apps/meteor/client/components/message/variants/RoomMessage.tsx [hunk 2]
```diff
@@ -41,6 +41,7 @@ const RoomMessage = ({ message, sequential, all, mention, unread, context, ignor
 	const chat = useChat();
 	const messageRef = useRef(null);
 
+	const showUserAvatar = useUserPreference<boolean>('displayAvatars');
```
Comment (ggazzo) [inline apps/meteor/client/components/message/variants/RoomMessage.tsx:44]: would you mind passing that value via prop?
I'm afraid this hook might add a bit of burden to rendering as it's the same for all list items, maybe you could use it in MessageList.tsx

**Review by ggazzo — CHANGES_REQUESTED** [review]
please check the comment and apply to all cases

File: apps/meteor/client/components/message/variants/RoomMessage.tsx (hunk 2 above)
Comment (nishant23122000) [reply apps/meteor/client/components/message/variants/RoomMessage.tsx:44]: Sure @ggazzo

Comment (nishant23122000) [reply apps/meteor/client/components/message/variants/RoomMessage.tsx]: Please check now.

**Review by ggazzo — APPROVED** [review]

**Pull request merged**